commit 4ad23a43add7c9d75283e99c5beb14ff7babe54a (refs/changes/00/4700/4)
Author: Dave Eddy <dave@daveeddy.com>
Date:   2018-08-17T13:20:19-04:00 (1 year, 2 months ago)
    
    joyent/node-manta#303 mfind should probably require at least one path argument
    Reviewed by: Robert Mustacchi <rm@joyent.com>

diff --git a/bin/mfind b/bin/mfind
index 820582b..6d27942 100755
--- a/bin/mfind
+++ b/bin/mfind
@@ -104,11 +104,7 @@ function parseCmdOptions(opts, parser) {
         opts.name = new RegExp(opts.name);
     }
 
-    if (!opts._args.length) {
-        opts.paths = ['/' + opts.account + '/stor'];
-    } else {
-        opts.paths = opts._args;
-    }
+    opts.paths = opts._args;
 
     return (opts);
 }
@@ -156,6 +152,7 @@ function printEntry(opts, obj) {
         }
     });
 
+    assert(options.paths.length >= 1, 'no paths specified');
     options.paths.forEach(function (p) {
         barrier.start(p);
         client.ftw(p, options, function (err, res) {
diff --git a/test/mfind.test.js b/test/mfind.test.js
index bc77413..af573f4 100644
--- a/test/mfind.test.js
+++ b/test/mfind.test.js
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016 Joyent, Inc.
+ * Copyright 2018 Joyent, Inc.
  */
 
 /*
@@ -111,6 +111,19 @@ test('check if operator (mfind forbidden)', function (t) {
     });
 });
 
+/*
+ * node-manta#303 mfind should probably require at least one path argument
+ */
+test('mfind (no arguments)', function (t) {
+    forkExecWait({
+        argv: [MFIND]
+    }, function (err, info) {
+        t.ok(err, 'mfind should fail');
+        t.ok(/^path required/m.test(info.stderr), 'path required in stderr');
+        t.done();
+    });
+});
+
 test('mfind TESTDIR', function (t) {
     forkExecWait({
         argv: [MFIND, TESTDIR]
